### PR TITLE
Feature/process

### PR DESCRIPTION
**Pull request by tuj**
* Added process entity migrations
* Moved code out of App and into separate bundle: https://github.com/aakb/kontrolgruppen-core-bundle
* Added development setup and updated readme.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -17,11 +17,8 @@ Encore
      * Each entry will result in one JavaScript file (e.g. app.js)
      * and one CSS file (e.g. app.css) if you JavaScript imports CSS.
      */
-    .addEntry('app', './assets/js/app.js')
-    //.addEntry('page1', './assets/js/page1.js')
-    //.addEntry('page2', './assets/js/page2.js')
-
-    .addEntry('login', './assets/css/login.css')
+    .addEntry('core', './vendor/kontrolgruppen/core-bundle/assets/js/app.js')
```
Comment (rimi-itk) [inline webpack.config.js]: I don't like the reference to `vendor` here. Shouldn't the core-bundle provide assets so we don't have to do anything unless we want to change some assets in this project?

File: scripts/install.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+bold=$(tput bold)
+normal=$(tput sgr0)
+
+dir=$(cd $(dirname "${BASH_SOURCE[0]}")/../ && pwd)
+
+mkdir $dir/packages
+mkdir $dir/packages/kontrolgruppen
```
Comment (rimi-itk) [inline scripts/install.sh]: One-liner: `mkdir -p $dir/packages/kontrolgruppen`

File: scripts/develop.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        rm -rf $dir/vendor/$vendor/$package
+
+        COMPOSER=$dir/composer-dev.json php -d memory_limit=-1 $COMPOSER_PATH config repositories.$vendor/$package '{"type": "path", "url": "packages/'$vendor'/'$package'", "options": {"symlink": true}}'
+    done
+done
+
+# Install packages.
+cd $dir
+COMPOSER=$dir/composer-dev.json php -d memory_limit=-1 $COMPOSER_PATH install
+bin/console assets:install --symlink --relative
```
Comment (rimi-itk) [inline scripts/develop.sh]: This is already done in https://github.com/aakb/kontrolgruppen/blob/develop/composer.json#L68

File: scripts/develop.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
```
Comment (rimi-itk) [inline scripts/develop.sh]: I think we should move the develop(ment) scripts in to `scripts/develop` to make it clear that they're only used for development. Then we can at some point add general install/update scripts, say, in the `scripts` folder.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,23 @@
 # Kontrolgruppen
 
+## Setup for development
```
Comment (rimi-itk) [inline README.md]: Maybe we should move this into a `README.develop.md` (or something like that) to keep `README.md` clean and only contain information needed for getting the project up and running.

File: README.md [hunk 6]
```diff
@@ -1,5 +1,23 @@
...
+
+```sh
+# Get the kontrolgruppen bundles to packages/.
+/scripts/install.sh
```
Comment (rimi-itk) [inline README.md]: Should be `./scripts/install.sh`.

File: README.md [hunk 7]
```diff
@@ -1,5 +1,23 @@
...
+
+# Create composer-dev.json and install the local kontrolgruppen/* bundles.
+/scripts/develop.sh
```
Comment (rimi-itk) [inline README.md]: Should be `./scripts/develop.sh`.

File: README.md [hunk 8]
```diff
@@ -1,5 +1,23 @@
...
+
+bin/console doctrine:migrations:migrate
+```
+
+## Requiring new bundles
+
+Should be required in the `kontrolgruppen/core-bundle` bundle instead.
```
Comment (rimi-itk) [inline README.md]: Well, it depends on if the bundles are generally neede or only needed in this project …

File: scripts/develop.sh (hunk 3 above)
Comment (rimi-itk) [reply scripts/develop.sh]: After running `./scripts/develop.sh`:

```sh
$ git status
…
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	assets/
	src/Repository/
	templates/base.html.twig
```
Can we prevent this?

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: composer.json [hunk 9]
```diff
@@ -1,12 +1,14 @@
...
     "license": "MIT",
-    "minimum-stability": "stable",
+    "minimum-stability": "dev",
+    "prefer-stable":true,
     "require": {
         "php": "^7.1.3",
         "ext-ctype": "*",
         "ext-iconv": "*",
-        "friendsofsymfony/user-bundle": "^2.1",
+        "kontrolgruppen/core-bundle": "dev-develop",
```
Comment (rimi-itk) [inline composer.json]: This should probably be `…: "^1.0"` …

File: scripts/develop.sh [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    echo $vendor
+
+    for package_dir in $vendor_dir/*/ ; do
+        package=$(basename "$package_dir")
+        echo - $package
+
+        rm -rf $dir/vendor/$vendor/$package
+
+        COMPOSER=$dir/composer-dev.json php -d memory_limit=-1 $COMPOSER_PATH config repositories.$vendor/$package '{"type": "path", "url": "packages/'$vendor'/'$package'", "options": {"symlink": true}}'
+    done
```
Comment (rimi-itk) [inline scripts/develop.sh]: Are we missing `composer require $vendor/$package:dev-develop` here? We want to require the `develop` branch from the packages, right?

File: webpack.config.js (hunk 1 above)
Comment (tuj) [reply webpack.config.js]: This is definitely a problem area with the setup.
I have not been able to fix the webpack setup yet.
I'll add a comment to remember to fix it.

File: README.md (hunk 8 above)
Comment (tuj) [reply README.md]: True. I'll remove the section.

File: composer.json (hunk 9 above)
Comment (tuj) [reply composer.json]: I will do that when there is a 1.0.0 version of the bundle

File: scripts/develop.sh (hunk 10 above)
Comment (tuj) [reply scripts/develop.sh]: Yes

File: README.md (hunk 5 above)
Comment (tuj) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (tuj) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (tuj) [reply README.md]: Done

File: scripts/develop.sh (hunk 4 above)
Comment (tuj) [reply scripts/develop.sh]: Done

File: scripts/develop.sh (hunk 3 above)
Comment (tuj) [reply scripts/develop.sh]: Yes

File: scripts/install.sh (hunk 2 above)
Comment (tuj) [reply scripts/install.sh]: Done

File: composer.json (hunk 9 above)
Comment (tuj) [reply composer.json]: Changed to dev-master until a tag i ready, which will be changed to dev-develop by the develop.sh script.

File: webpack.config.js (hunk 1 above)
Comment (tuj) [reply webpack.config.js]: Removed from project.
Webpack build setup added to core-bundle

**Pull request merged**